flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/JobResultStoreOptions.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.highavailability;
+
+import org.apache.flink.annotation.docs.Documentation;
+import org.apache.flink.configuration.ConfigOption;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/** The set of configuration options relating to the Job Result Store. */
+public class JobResultStoreOptions {
+    @Documentation.Section(Documentation.Sections.COMMON_HIGH_AVAILABILITY)
+    public static final ConfigOption<String> STORAGE_PATH =
+            key("job-result-store.storage-path")
+                    .stringType()
+                    .defaultValue(null)
+                    .withDescription(
+                            "Defines where job results should be stored. This should be an "
+                                    + "underlying file-system that provides read-after-write consistency. By "
+                                    + "default, uses a combination of high-availability.storageDir and "
+                                    + "high-availability.cluster-id to determine the path.");
```

Comment:
I have! Would there be an objection to just moving the config options wholesale into the FilesystemJobResultStore class? If they are not used anywhere else, I feel like having a seperate class is not so necessary and it keeps the documentation closer to the implementation, as you mentioned.

Reply:
@XComp Ah, it doesnt look like this works. I think when it generates the docs it looks in a specific package, and isnt picked up if its in the `FilesystemJobResultStore` class. What do you mean about "the description (not the entire text)"? I feel like having text in 2 places might be confusing - maybe it would be better to remove implementation details at all?